scipy-2023-data-grammar-poster/poster.rmd

```diff
@@ -0,0 +1,125 @@
+</script>
+
+<pre class="mermaid">
+flowchart LR
+
+    images[(Cell Images)]
+    csv[(CSV Files)]
+    npz[(NPZ Files)]
+    sqlite[(SQLite Files)]
+    extract["Feature\nextraction"]
+    challenges{{"- Large scale\n- Schema diff. \n- API diff."}}
+
+    images --> extract
+    extract --> csv --> challenges
+    extract --> sqlite --> challenges
+    extract --> npz --> challenges
+
+    style challenges fill:#FDA4AF,stroke:#881337
+</pre>
+<p class="figcaption">Figure 1. A diagram showing many different feature data and common challenges.</p>
+
+Research in the [Way Lab](https://www.waysciencelab.com/) involves intensive data engineering over high-dimensional single-cell morphology data from large-scale microscopy drug screening applications.
+Software development surrounding this work often entails scalability (larger than memory data handling) and understandability (syntax complexity and software sustainability) challenges.
+
+# Solution design
+
+<pre class="mermaid">
+flowchart LR
+
+    direction LR
+    data[(data)]
+    syntax{syntax}
+    output[/output/]
+
+    data -.- |+| syntax -.- |=| output
+    
+    linkStyle 0 stroke:#dddddd,color:black;
+    linkStyle 1 stroke:#dddddd,color:black;
+</pre>
+<p class="figcaption">Figure 2. A diagram illustrating data grammar as an abstract linguistic algorithm.</p>
+
+To address these challenges we have developed a python package called [CytoTable](https://github.com/cytomining/CytoTable) which implements "cross-language data grammar" capabilities (vocabulary + syntax = output) orchestrated with [Parsl](https://github.com/Parsl/parsl) workflows.
+Our vision is for CytoTable to increase consistency and reliability of data and enable more scientists to quickly access single-cell insights from microscopy images.
+
+<br>
+<br>
+
+##  <i class="fa fa-circle fa-lg"></i> Data: Apache Arrow
+
+![Figure 3. Chart showing relative memory size for data using various Python libraries.](https://raw.githubusercontent.com/cytomining/CytoTable-benchmarks/init-benchmarks/notebooks/images/mem-read-size.png)
+
+[Apache Arrow](https://arrow.apache.org/) represents a new frontier for data implementation flexibility, enabling a unified, multi-language, zero-copy format for in-memory analysis.
+Arrow is like a high-performance Pandas dataframe which may be used across languages with fewer scalability challenges.
+
+## <i class="fa fa-stream fa-lg"></i> Syntax: DuckDB SQL
+
+![Figure 4. Chart showing read time durations of Pandas and DuckDB with SQLite databases of various sizes.](https://raw.githubusercontent.com/cytomining/CytoTable-benchmarks/init-benchmarks/notebooks/images/join-read-time.png)
+
+Structured Query Language (SQL) through [DuckDB](https://duckdb.org/) provides an Arrow-compatible embedded database system optimized for vectorized execution.
+DuckDB delivers in-memory capabilities through SQL, treating variable data as a loose collection of database tables without needing conversion.
```

Comment:
"DuckDB delivers in-memory capabilities through SQL" could be interpreted as SQL somehow enabling these in-memory capabilities, but I think you mean that with DuckDB you get both in-memory data manipulation and a SQL interface as two separate things. Perhaps rewriting this as "DuckDB delivers in-memory data manipulation capabilities through a SQL interface" would be more clear?

Reply:
Great catch, yes, this is better and more correct to the functionality. I'll add this in.